posts/2021-08-30-binding-app-to-postgresql-db-with-odo-and-ol-stack.adoc

```diff
@@ -0,0 +1,284 @@
+
+== Service Binding Operator
+
+The link:https://github.com/redhat-developer/service-binding-operator/blob/master/README.md[Service Binding Operator] makes it easier for developers to bind operator managed services (i.e. databases) to applications. More specifically, the operator will collect service data, deploy resources (i.e. secrets, configmaps) containing that data, and will make the data available to the application through the ServiceBinding custom resource it provides.
+
+== Try it out
+
+To begin, you need three things:
+
+- Kubernetes cluster. For this blog, an OpenShift cluster is used. Be sure to log in. Odo will interact with your kubernetes cluster.
+- OpenShift Do. If you have not already installed odo, do so now by following the instructions outlined in the link:https://odo.dev[odo] documentation. Be sure to install version 2.2.4 and above.
+- An application that uses a PostgreSQL database. For this blog, the Open Liberty stack link:https://github.com/OpenLiberty/application-stack-samples/tree/main/jpa[sample JPA application] is used.
+
+
+=== Install the needed operators
+
+If you have the right privileges, you can use the OpenShift console to install these operators. You do this by navigating to `Operators->OperatorHub` and searching/selecting/installing a couple of operators from the catalog.
+
+- Service Binding Operator (provided by Red Hat)
+
+Use the OpenShift console (`Operators->OperatorHub`) to install this operator. Be sure to install version 0.9.1 and above.
+
+- PostgreSQL Operator (provided by Dev4Ddevs.com)
+
+Before installing this operator, create the project/namespace under which the application will be deployed. This operator is namespace scoped, so it needs to be installed in the same namespace where the application will be deployed.
+
+[source,sh]
+----
+odo project create service-binding-demo
+----
+
+Use the OpenShift console (`Operators->OperatorHub`) to install this operator. Be sure to pick `service-binding-demo` as the namespace in which to install it.
+
+=== Create a Java Open Liberty based component and the Database resource service
+
+Clone the application repository.
+
+[source,sh]
+----
+git clone https://github.com/OpenLiberty/application-stack-samples.git && \
+cd application-stack-samples/jpa
+----
+
+Create a Java Open Liberty component.
+
+[source,sh]
+----
+odo create java-openliberty mysboproj
+----
+
+Display the service providers and services available in the cluster.
+
+[source,sh]
+----
+odo catalog list services
+----
+
+Output:
+
+[source,sh]
+----
+Services available through Operators
+NAME                                CRDs
+...
+postgresql-operator.v0.1.1          Backup, Database
+...
+----
+
+Generate the `postgresql-operator.v0.1.1` provided Database resource/service yaml and store it in a file. The data is stored in a file because some entries such as the resource name, the access information, and annotations will be customized.
```

Comment:
Generate the `postgresql-operator.v0.1.1` database resource and store it in a yaml file. The data is stored in a file because some entries will be customized, such as the resource name, the access information, and annotations.